exhibitions/src/main/kotlin/edu/artic/exhibitions/ExhibitionDetailFragment.kt

```diff
@@ -0,0 +1,202 @@
+                                            val animator = ValueAnimator.ofInt(parentWidth, newHeight)
+                                            animator.addUpdateListener {
+                                                val params = exhibitionImage.layoutParams
+                                                params.width = parentWidth
+                                                params.height = it.animatedValue as Int
+                                                exhibitionImage.layoutParams = params
+                                            }
+                                            animator.duration = 500
+                                            animator.start()
+                                        }
+
+                                    }
+                                    startPostponedEnterTransition()
+                                    return false
+                                }
+
+                            }).into(exhibitionImage)
+                }
+                .disposedBy(disposeBag)
+
+        viewModel.metaData
+                .bindToMain(metaData.text())
+                .disposedBy(disposeBag)
+
+        viewModel.description
+                .bindToMain(description.text())
+                .disposedBy(disposeBag)
+
+        viewModel.throughDate
+                .bindToMain(throughDate.text())
+                .disposedBy(disposeBag)
+
+        viewModel.showOnMapButtonText
+                .bindToMain(showOnMap.text())
+                .disposedBy(disposeBag)
+
+        viewModel.buyTicketsButtonText
+                .bindToMain(buyTickets.text())
+                .disposedBy(disposeBag)
+
+        showOnMap.clicks()
+                .subscribe { viewModel.onClickShowOnMap() }
+                .disposedBy(disposeBag)
+
+        buyTickets.clicks()
+                .subscribe { viewModel.onClickBuyTickets() }
+                .disposedBy(disposeBag)
+
+    }
+
+    override fun setupNavigationBindings(viewModel: ExhibitionDetailViewModel) {
+        viewModel.navigateTo
+                .subscribe {
+                    when (it) {
+                        is Navigate.Forward -> {
+                            when (it.endpoint) {
+                                is ExhibitionDetailViewModel.NavigationEndpoint.ShowOnMap -> {
+                                    Timber.d("Show on map")
+                                }
+
+                                is ExhibitionDetailViewModel.NavigationEndpoint.BuyTickets -> {
+                                    val endpoint = it.endpoint as ExhibitionDetailViewModel.NavigationEndpoint.BuyTickets
+                                    var url = endpoint.url
+                                    if (!url.startsWith("http://") && !url.startsWith("https://"))
+                                        url = "https://$url"
+                                    val myIntent = Intent(Intent.ACTION_VIEW, Uri.parse(url))
```

Comment:
helper class. Sameer @sam33rdhakal I had same comment. so put them in the same place maybe.